# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../epeg/example-scale.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

The example applications is magnitudes more useful if it allows to specify
the scale factor ,-)

  - Rene Rebe <rene@exactcode.de>

--- epeg/src/bin/epeg_main.c	2000-01-01 04:00:00.000000000 +0100
+++ epeg-feature/src/bin/epeg_main.c	2005-07-13 11:37:54.000000000 +0200
@@ -6,22 +6,31 @@
 main(int argc, char **argv)
 {
    Epeg_Image *im;
+   int w, h;
+   int scale = 50;
 
-   if (argc != 3)
+   if (argc != 4)
      {
-	printf("Usage: %s input.jpg thumb.jpg\n", argv[0]);
-	exit(0);
+	printf("Usage: %s n%% input.jpg thumb.jpg\n", argv[0]);
+	exit(1);
      }
-   im = epeg_file_open(argv[1]);
+
+   scale = atoi (argv[1]);
+   if (scale <= 0 || scale >= 100)
+     {
+       printf("Scale not between 0..100\n");
+       exit(1);
+     }
+
+   im = epeg_file_open(argv[2]);
    if (!im)
      {
-	printf("cannot open %s\n", argv[1]);
+	printf("cannot open %s\n", argv[2]);
 	exit(-1);
      }
      {
 	const char *com;
 	Epeg_Thumbnail_Info info;
-	int w, h;
 	
 	com = epeg_comment_get(im);
 	if (com) printf("Comment: %s\n", com);
@@ -38,7 +47,8 @@
 	printf("Image size: %ix%i\n", w, h);
      }
    
-   epeg_decode_size_set           (im, 128, 96);
+   epeg_decode_size_set           (im, w*scale/100, h*scale/100);
+   printf("Decode size: %ix%i\n", w*scale/100, h*scale/100);
 
 #if 0   
      {
@@ -69,8 +79,8 @@
    
    epeg_quality_set               (im, 80);
    epeg_thumbnail_comments_enable (im, 1);
-   epeg_comment_set               (im, "Smelly pants!");
-   epeg_file_output_set           (im, argv[2]);
+   epeg_comment_set               (im, "epeg thumbnail");
+   epeg_file_output_set           (im, argv[3]);
    epeg_encode                    (im);
    epeg_close                     (im);
    return 0;
